src/pages/main/MainPage.tsx

```diff
@@ -0,0 +1,36 @@
+import cls from './mainPage.module.scss';
+import CardsHandler from '../../components/Cards/CardsHandler';
+import { useState, useEffect, useCallback } from 'react';
+import { useSearchParams } from 'react-router-dom';
+import { MainPageRoutes } from '../../app/providers/router/routeConfig/routeConfig';
+import { CardsHandlerSize } from '../../components/Cards/CardsHandler';
+import { Outlet } from 'react-router-dom';
+
+export const MainPage = () => {
+  const [searchParams, setSearchParams] = useSearchParams();
+  const [isDetailedOpen, setIsDetailedOpen] = useState<boolean>(false);
+  const isShowOpen = searchParams.get(MainPageRoutes.SHOW);
+
+  useEffect(() => {
+    setIsDetailedOpen(!!isShowOpen);
```

Comment:
It's bad practice to use double logical NOT operator.
Instead of this use please `Boolean()`